corehq/apps/cleanup/management/commands/populate_sql_model_from_couch_model.py

```diff
@@ -53,12 +53,24 @@ def diff_couch_and_sql(cls, couch, sql):
         """
         raise NotImplementedError()
 
+    @classmethod
+    def get_filtered_diffs(cls, couch, sql):
+        diffs = cls.diff_couch_and_sql(couch, sql)
+        if isinstance(diffs, list):
+            diffs = list(filter(None, diffs))
+        return diffs
+
+    @classmethod
+    def get_diff_as_string(cls, couch, sql):
+        diffs = cls.get_filtered_diffs(couch, sql)
+        return "\n".join(diffs) if diffs else None
+
     @classmethod
     def diff_attr(cls, name, doc, obj, wrap_couch=None, wrap_sql=None, name_prefix=None):
         """
         Helper for diff_couch_and_sql
         """
-        couch = doc.get(name, None)
+        couch = doc.get(name, None) if doc else None
```

Comment:
Do you think there will be a need for this in other migrations? I'd have been inclined to do `couch_permission._asdict() if couch_permission else {}` below rather than changing this, just because when reading this code it's surprising to think of `doc` not existing.

Reply:
I guess that's fair though I I don't see that adding null safety is ever a bad thing.

Reply:
c7cd3ce7426

Reply:
I strongly agree with Jenny on this one. Excuse me for butting in here and climbing on my soap box.

> I don't see that adding null safety is ever a bad thing.

I disagree with this. It's a long-standing pet peeve of mine since this is done in many places throughout the HQ codebase. Adding "null safety" necessitates all kinds of extra null checks and code that tries to short-circuit around areas that require real values. It leaks out everywhere, and it makes code harder to reason about because there are more possible "valid" states and code paths to be considered. I think it's almost always better to write code in a way that a given variable can never be null, and write tests for the edge cases where a null value might show up to ensure that scenario is handled.

I see you've fixed it in https://github.com/dimagi/commcare-hq/pull/29688/commits/c7cd3ce742690b4888d2569034b98fd785b0d234, so thank you for that.